lib/linguist/languages.yml

```diff
@@ -3176,6 +3176,7 @@ JSON with Comments:
   - ".jshintrc"
   - ".jslintrc"
   - ".swcrc"
+  - ".vscode/settings.json"
```

Comment:
This won't work. Linguist doesn't consider the directory components of file paths when examining filenames.

Reply:
Bummer. I thought I was taking a chance when I didn't see any others like this. Would it be reasonable to just put `settings.json`? It doesn't seem like that distinctive a name, but maybe it is.

Reply:
> Would it be reasonable to just put `settings.json`?

Nope, sorry. That filename is _waaaay_ too generic. Best solution is to add this to your project's `.gitattributes` file (create one if necessary):

~~~gitattributes
.vscode/**/*.json linguist-language=JSONC
~~~

Reply:
Thank you, I'll do exactly that.

Reply:
Looking forward, which of these possibilities seem more plausible?

- Linguist considers filepaths, not just filenames. 
- VS Code switches to more distinctively named dot files, e.g. `settings.json` -> `.vscodesettings`.

Reply:
Neither strikes me as particularly plausible; the former would involve a complete rehaul of Linguist's codebase, while the latter would only become eligible once there's sufficient distribution of `.vscodesettings` files across GitHub (see [`CONTRIBUTING.md`](https://github.com/github-linguist/linguist/blob/master/CONTRIBUTING.md#adding-a-language) for rationale and specifics).

Instead, it'd be better if VS Code adopted `.jsonc` as a file extension (which Linguist [already supports](https://github.com/github-linguist/linguist/blob/master/lib/linguist/languages.yml#L3156)) instead of just `.json` (which is too easily conflated with proper JSON).

Reply:
Great idea!

Reply:
https://github.com/microsoft/vscode/issues/191988